src/lib.rs

```diff
@@ -307,6 +307,31 @@ impl Map {
         }
         maximum_ts
     }
+
+    /// Computes the rectangle on the image where the sprite is stored for the given tile ID.
+    /// If the ID is not found in any tileset, or if there is no image associated with the tileset, `None` is returned.
+    /// On success, returns `Some(x, y, w, h)`, where `(x, y)` is the coordinates of the top-left corner, and `(w, h)` are the width and height of the rectangle
+    pub fn get_tile_rectangle_by_id(&self, id: u32) -> Option<(u32, u32, u32, u32)> {
+        let tileset = self.get_tileset_by_gid(id)?;
+        let img = tileset.images.get(0)?; // we suppose there is only 1 image per tileset
+
+        let id = id - 1;
+        let columns =
+            (img.width as u32 - 2 * tileset.margin) / (tileset.spacing + tileset.tile_width);
```

Comment:
That's correct, but rather than subtracting the tile width and adding one, we can instead just add one spacing to the image width.

And actually, the margin is only used on the left and top of the tileset. It does not prevent tiles from getting cut out from the bottom or right sides.

So in the end, this is the calculation used by Tiled:

```suggestion
            (img.width as u32 - tileset.margin + tileset.spacing) / (tileset.tile_width + tileset.spacing);
```